src/velocitas/model_generator/tree_generator/file_formats.py

```diff
@@ -36,22 +36,33 @@ def load_tree(self):
 
 class Vspec(FileFormat):
     def __init__(
-        self, file_path: str, include_dirs: List, strict: bool, overlays: List[str]
+        self,
+        file_path: str,
+        unit_file_path_list: List[str],
+        include_dirs: List,
+        strict: bool,
+        overlays: List[str],
     ):
         super().__init__(file_path)
+        self.unit_file_path_list = unit_file_path_list
         self.include_dirs = include_dirs
         self.strict = strict
         self.overlays = overlays
 
     def load_tree(self):
+        """loads a tree of a vspec file through vss-tools"""
         print("Loading vspec...")
+        vspec.load_units(
+            self.file_path,
+            self.unit_file_path_list,
+        )
         tree = vspec.load_tree(
             self.file_path,
             self.include_dirs,
-            merge_private=False,
-            break_on_unknown_attribute=self.strict,
+            tree_type=vspec.VSSTreeType.SIGNAL_TREE,
             break_on_name_style_violation=self.strict,
             expand_inst=False,
+            data_type_tree=vspec.VSSTreeType.SIGNAL_TREE,
```

Comment:
where is the difference between tree_type and data_type_tree?

Reply:
it's a function from vss-tools and it defines the type with the following: https://github.com/COVESA/vss-tools/blob/f7026ba21d2974106ea7604a472701fe5060f533/vspec/model/constants.py#L278
this means that you get different trees with different attributes. As of now no one is using data_type_tree. It would get used if you use structs. So yes let's ignore this and set it to None.

Reply:
it already defaults to None (data_type_tree: Optional[VSSNode] = None), so you can just remove it